binaries/oli/src/config/mod.rs

```diff
@@ -0,0 +1,162 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use opendal::{services, Operator, Scheme};
+use serde::Deserialize;
+use toml;
+
+type StringMap<T> = HashMap<String, T>;
+
+macro_rules! update_options {
+    ($m: expr) => {{
+        let mut opts = $m.clone();
+        opts.remove("type");
+        opts
+    }};
+}
+
+#[derive(Deserialize, Default)]
+pub struct Config {
+    profiles: StringMap<StringMap<String>>,
+}
+
+impl Config {
+    /// Parse a local config file.
+    ///
+    /// - If the config file is not present, a default Config is returned.
+    pub fn load_from_file<P: AsRef<Path>>(fp: P) -> Result<Config> {
+        let config_path = fp.as_ref();
+        if !config_path.exists() {
+            return Ok(Config::default());
+        }
+        let data = fs::read_to_string(config_path)?;
+        Config::load_from_str(&data)
+    }
+
+    pub(crate) fn load_from_str(s: &str) -> Result<Config> {
```

Comment:
I prefer to merge this function if it's function only be used once.

Reply:
The purpose of this function is to make testing the config parser easy, otherwise we need to create a real file in the unit test.